sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClient.cs

```diff
@@ -1222,5 +1224,119 @@ public virtual async Task<Response<byte[]>> GetRandomBytesAsync(int count, Cance
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Releases a key.
+        /// </summary>
+        /// <param name="name">The name of the key to release.</param>
+        /// <param name="target">The attestation assertion for the target of the key release.</param>
+        /// <param name="options">Optional <see cref="ReleaseKeyOptions"/> containing additional options to release a key.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <remarks>
+        /// The key must be exportable.
+        /// This operation requires the keys/release permission.
+        /// </remarks>
+        /// <returns>The key release result containing the released key.</returns>
+        /// <exception cref="ArgumentException"><paramref name="name"/> or <paramref name="target"/> contains an empty string.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="name"/> or <paramref name="target"/> is null.</exception>
+        /// <exception cref="RequestFailedException">The server returned an error. See <see cref="Exception.Message"/> for details returned from the server.</exception>
+        public virtual Task<Response<ReleaseKeyResult>> ReleaseKeyAsync(string name, string target, ReleaseKeyOptions options = default, CancellationToken cancellationToken = default) =>
+            ReleaseKeyAsync(name, null, target, options, cancellationToken);
+
+        /// <summary>
+        /// Releases a key.
+        /// </summary>
+        /// <param name="name">The name of the key to release.</param>
+        /// <param name="version">Optional version of the key to release.</param>
+        /// <param name="target">The attestation assertion for the target of the key release.</param>
+        /// <param name="options">Optional <see cref="ReleaseKeyOptions"/> containing additional options to release a key.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <remarks>
+        /// The key must be exportable.
+        /// This operation requires the keys/release permission.
+        /// </remarks>
+        /// <returns>The key release result containing the released key.</returns>
+        /// <exception cref="ArgumentException"><paramref name="name"/> or <paramref name="target"/> contains an empty string.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="name"/> or <paramref name="target"/> is null.</exception>
+        /// <exception cref="RequestFailedException">The server returned an error. See <see cref="Exception.Message"/> for details returned from the server.</exception>
+        public virtual async Task<Response<ReleaseKeyResult>> ReleaseKeyAsync(string name, string version, string target, ReleaseKeyOptions options = default, CancellationToken cancellationToken = default)
```

Comment:
Is the overload with the `version` parameter idiomatic to dotnet? We don't usually do that in JS so I was curious whether that is necessary or if `version` should just be in the options bag (like we have it)

Reply:
In a previous archboard review it was decided to keep `name` and `version` parameters first and in that order as seen in other methods. At least in .NET, you can't have required parameters i.e. `target` after an optional parameter. As noted elsewhere, `version` is actually optional - not required.

I don't know if this is true of Python or JS, but I imagine at least with JS you'd have to do something similar. Java will have to do overloads as well since they don't even support optional parameters.

Reply:
Not sure if this would be a problem with Python since they typically use named parameters.